### PR TITLE
Update 1.7.4-aws docs

### PR DESCRIPTION
**Pull request by bwbarrett**
Update docs to reflect all the changes since 1.7.3-aws.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: RELEASENOTES.md [hunk 1]
```diff
@@ -14,6 +14,51 @@ release) was only available in one of the two variants, we note that
...
 
 
+# v1.7.4-aws release notes
+This release requires [Libfabric v1.11.0](https://github.com/ofiwg/libfabric/releases/tag/v1.11.0)
+or later and supports [NCCL v2.19.3-1](https://github.com/NVIDIA/nccl/releases/tag/v2.19.3-1) while
+maintaining backward compatibility with older NCCL versions ([NCCL v2.4.8](https://github.com/NVIDIA/nccl/releases/tag/v2.4.8-1) and later).
+It was tested with Libfabric versions up to
+[Libfabric v1.19.0](https://github.com/ofiwg/libfabric/releases/tag/v1.19.0).
+
+With NCCL 2.18.5 or later and v1.7.3-aws of the plugin,
```
Comment (rauteric) [inline RELEASENOTES.md:24]: v1.7.3-aws or later? Do we still need this paragraph at all? It is the same in the 1.7.3-aws release notes.

File: RELEASENOTES.md [hunk 2]
```diff
@@ -14,6 +14,51 @@ release) was only available in one of the two variants, we note that
...
+if NCCL 2.18.5 or later is used, users may wish to disable it by
+setting `NCCL_NVLS_ENABLE=0` in the environment of your job.
+
+New Features:
+* Hard fail if GPUDirect RDMA initialization fails on an EC2 instance
+  that should support GPUDirect RDMA (such as P4d.24xlarge or
+  P5.48xlarge), rather than fall back to host copy buffers at
+  significantly reduced performance.  Setting the environment variable
+  `OFI_NCCL_DISABLE_GDR_REQUIRED_CHECK=1` will disable this behavior.
+* Change the threshold at which the rdma provider switches from round
```
Comment (rauteric) [inline RELEASENOTES.md:40]: Nit: RDMA protocol / RDMA transport

File: RELEASENOTES.md (hunk 1 above)
Comment (bwbarrett) [reply RELEASENOTES.md]: fixed

File: RELEASENOTES.md (hunk 2 above)
Comment (bwbarrett) [reply RELEASENOTES.md]: fixed

**Review by rauteric — APPROVED** [review]

**Pull request merged**